elastic/kibana
# [Alerting] Updating ES query rule to correctly handle epoch millis timestamps

Resolves https://github.com/elastic/kibana/issues/104324

## Summary

Missing a `format` field inside a range filter in the ES query that would allow the query to run on `epoch_millis` mapped time fields. Added a functional test to cover this case.

## To verify
You can comment out the change and run the functional test, see it error with this message `An error occurred when running the rule. search_phase_execution_exception, caused by: "failed to parse date field [2021-07-05T15:06:59.918Z] with format [epoch_millis],date_time_parse_exception: Failed to parse with all enclosed parsers"`, and then see it succeed with the fix.

or

Create an ES query rule on an index with a timestamp field that is mapped with format `epoch_millis`. Verify that the rule runs correctly with no errors after becoming `Active`. The error doesn't show up if the rule never becomes `Active` so you will need to ensure that the rule becomes `Active` at least once and that subsequent runs succeed without erroring.

### Checklist

Delete any items that are not applicable to this PR.
